core/storage/topic/gae_models_test.py

```diff
@@ -0,0 +1,167 @@
+# limitations under the License.
+
+"""Tests for Topic model."""
+
+from constants import constants
+from core.domain import topic_domain
+from core.domain import topic_services
+from core.platform import models
+from core.tests import test_utils
+import feconf
+
+(topic_models,) = models.Registry.import_models([models.NAMES.topic])
+
+
+class TopicModelUnitTests(test_utils.GenericTestBase):
+    """Tests the TopicModel class."""
+
+    def test_that_subsidiary_models_are_created_when_new_model_is_saved(self):
+        """Tests the _trusted_commit() method."""
+
+        # Topic is created but not committed/saved.
+        topic = topic_models.TopicModel(
+            id='topic_id',
+            name='topic_name',
+            subtopic_schema_version=feconf.CURRENT_SUBTOPIC_SCHEMA_VERSION,
+            next_subtopic_id=1,
+            language_code='en'
+        )
+        # We check that topic has not been saved before calling commit().
+        self.assertIsNone(topic_models.TopicModel.get_by_name('topic_name'))
+        # We call commit() expecting that _trusted_commit works fine
+        # and saves topic to datastore.
+        topic.commit(
+            committer_id=feconf.SYSTEM_COMMITTER_ID,
+            commit_message='Created new topic',
+            commit_cmds=[{'cmd': topic_domain.CMD_CREATE_NEW}]
+        )
+        # Now we check that topic is not None and that actually
+        # now topic exists, that means that commit() worked fine.
+        self.assertIsNotNone(topic_models.TopicModel.get_by_name('topic_name'))
+
+    def test_get_by_name(self):
+        topic = topic_domain.Topic.create_default_topic(
+            topic_id='topic_id',
+            name='topic_test'
+        )
+        topic_services.save_new_topic(feconf.SYSTEM_COMMITTER_ID, topic)
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(topic.name).name,
+            topic.name
+        )
+        self.assertEqual(
+            topic_models.TopicModel.get_by_name(topic.name).id,
+            topic.id
+        )
+
+
+class TopicCommitLogEntryModelUnitTest(test_utils.GenericTestBase):
+    """Tests the TopicCommitLogEntryModel class."""
+
+    def test__get_instance_id(self):
+        # Calling create() method calls _get_instance (a protected method)
+        # and set the instance id equal to the result of calling that method.
+        topic_commit_log_entry = (
+            topic_models.TopicCommitLogEntryModel.create(
+                entity_id='entity_id',
+                version=1,
+                committer_id='committer_id',
+                committer_username='committer_username',
+                commit_type='create',
+                commit_message='Created new TopicCommitLogEntry',
+                commit_cmds=[{'cmd': 'create_new'}],
+                status=constants.ACTIVITY_STATUS_PRIVATE,
+                community_owned=True
+            )
+        )
+        topic_commit_log_entry_test_instance_id = 'topic-%s-%s' % (
```

Comment:
Hardcode this fully: EXPECTED_TOPIC_COMMIT_LOG_ENTRY_ID. See comments above about using primitive values with no logic (for expected values).